packages/vscode-extension/src/handlers.ts

```diff
@@ -901,61 +901,96 @@ export async function grantPermission(env: string): Promise<Result<Void, FxError
   return result;
 }
 
-export async function listCollaborator(env: string): Promise<TreeItem[]> {
-  let result: TreeItem[] = [];
+export async function listAllCollaborators(envs: string[]): Promise<Record<string, TreeItem[]>> {
+  const result: Record<string, TreeItem[]> = {};
   ExtTelemetry.sendTelemetryEvent(TelemetryEvent.ListCollaboratorStart);
 
-  try {
-    const checkCoreRes = checkCoreNotEmpty();
-    if (checkCoreRes.isErr()) {
-      throw checkCoreRes.error;
-    }
+  let fatalError;
+  const checkCoreRes = checkCoreNotEmpty();
+  if (checkCoreRes.isErr()) {
+    fatalError = checkCoreRes.error;
+  }
 
-    const inputs: Inputs = getSystemInputs();
-    inputs.env = env;
-    const userList = await core.listCollaborator(inputs);
-    if (userList.isErr()) {
-      throw userList.error;
-    }
+  const inputs: Inputs = getSystemInputs();
+  // Todo, remove this when we have better solution not to select env
+  inputs.env = envs[0];
```

Comment:
Seems we don't need this line?

Reply:
Thanks, will update it according to the discussion

Reply:
Updated